src/reporter.ts

```diff
@@ -108,7 +112,7 @@ export class Reporter {
   reportTestDone(name: string, testRunInfo: any): void {
     const hasError = !!testRunInfo.errs.length;
     const testItemId = this.testItems.find((item) => item.name === name).id;
-    const testCaseId = this.testData[testItemId];
+    const { testCaseId } = this.testItems.find((item) => item.id === testItemId);
```

Comment:
looks like you find same testItem two times, use destructuring instead
`const {id: testItemId, testCaseId } = this.testItems.find((item) => item.name === name)`